src/thor/bidirectional_astar.cc

```diff
@@ -1024,13 +1022,18 @@ void BidirectionalAStar::SetOrigin(GraphReader& graphreader,
                           static_cast<uint32_t>(edge.distance() + 0.5), cost.cost);
     }
 
-    // Set the initial not_thru flag to false. There is an issue with not_thru
+    // TODO(nils): is this really still necessary? could imagine it's a problem when we snap on
+    // a oneway road leading only to a not-thru edge and a restricted edge and we'd HAVE TO take
+    // that not_thru edge, go the full circle inside the not_thru region and come back on the
+    // not_thru's opposite to take the now allowed turn. easily tested!
+
+    // Set the initial not_thru_pruning to false. There is an issue with not_thru
     // flags on small loops. Set this to false here to override this for now.
-    edgelabels_forward_.back().set_not_thru(false);
+    edgelabels_forward_.back().set_not_thru_pruning(false);
 
-    pruning_disabled_at_origin_ = pruning_disabled_at_origin_ ||
-                                  !edgelabels_forward_.back().closure_pruning() ||
-                                  !edgelabels_forward_.back().not_thru_pruning();
+    pruning_disabled_at_origin_ =
+        pruning_disabled_at_origin_ || !edgelabels_forward_.back().closure_pruning() ||
+        directededge->not_thru() || !edgelabels_forward_.back().destonly_pruning();
```

Comment:
`destonly_pruning` is also a valid condition to extend the search in the other direction